src/routes/blog/post/chatbot-with-webllm-and-webgpu/+page.markdoc

```diff
@@ -0,0 +1,474 @@
+
+With this understanding of WebLLM and WebGPU, we can now start building!
+
+## Setting up the HTML for our AI chat app
+
+Before we write any JavaScript, we need a user interface. This will be the visible part of our app, the dropdown for selecting models, the chat area, and the input box.
+
+Here's the plan:
+
+- We'll create a container for everything
+- Add a select box for choosing the model
+- Include a progress bar for when the model is loading
+- Display the chat history
+- Create a form with a text input and a button to submit prompts
+
+To begin, create an `index.html` file and paste the following code inside it:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Browser LLM Chat Demo</title>
+    <style>
+      /* We'll add styling here. But later! */
+    </style>
+  </head>
+  <body>
+    <div class="app-container">
+      <h1>Chat with LLM (In your browser)</h1>
+
+      <div class="controls">
+        <select id="model-select">
+          <option value="SmolLM2-360M-Instruct-q4f32_1-MLC">
+            SmolLM2 360M (Very Small)
+          </option>
+          <option value="Llama-3.1-8B-Instruct-q4f32_1-MLC">
+            Llama 3.1 8B (Medium)
+          </option>
+          <option value="Phi-3.5-mini-instruct-q4f32_1-MLC">
+            Phi 3.5 Mini (Large)
+          </option>
+        </select>
+        <button id="load-model">Load Model</button>
+      </div>
+
+      <div class="chat-container">
+        <div id="output">Select a model and click "Load Model" to begin</div>
+
+        <div
+          id="progress-container"
+          class="progress-container"
+          style="display: none"
+        >
+          <div class="progress-bar">
+            <div id="progress-fill" class="progress-fill"></div>
+          </div>
+          <div id="progress-text" class="progress-text">0%</div>
+        </div>
+
+        <form id="chat-form" class="form-group">
+          <input id="prompt" placeholder="Type your question..." disabled />
+          <button type="submit" disabled>Send</button>
+        </form>
+      </div>
+    </div>
+  </body>
+</html>
+
+```
+
+In the HTML file, we've created a chat interface with controls for model selection and loading. The interface includes a chat output area, progress indicators, and an input form, which we need to let users interact with the AI model.
+
+### Model selection
+
+Notice that in the `div` with class `controls`, we have a `select` element for model selection and a `button` for loading the model. These models represent different size/capability tradeoffs: SmolLM2-360M is very small (~580MB) for low-resource devices, Llama-3.1-8B is medium-sized (~6GB) with good multilingual support, and Phi-3.5-mini is larger (~5.5GB) with strong reasoning capabilities. All can run directly in modern browsers with WebGPU support.
```

Comment:
When we say larger, do we mean size or context window? Because in size, Llama is larger than Phi-3.5 but it says otherwise

Reply:
Updated. For context window, they all seem to share a 4k value, but official info is unclear, so I've left that out.
I've added the table and use case